text/0045-nft-deposits-asset-hub.md

```diff
@@ -0,0 +1,246 @@
+
+The deposit should be calculated as a function of the number of existing collections with maximum
+DOT and stablecoin values limiting the amount. With asset rates available via the Asset Conversion
+pallet, the system could take the lower value required. A sigmoid curve would make sense for this
+application to avoid sudden rate changes, as in:
+
+$$ \frac{\mathrm{min(DOT deposit, stable deposit)} }{\mathrm{1 + e^{a - b * x}} }$$
+
+where the constant `a` moves the inflection to lower or higher `x` values, the constant `b` adjusts
+the rate of the deposit increase, and the independent variable `x` is the number of collections or
+items, depending on application.
+
+## Drawbacks
+
+Modifying deposit requirements necessitates a balanced assessment of the potential drawbacks.
+Highlighted below are cogent points extracted from the discourse on the [Polkadot Forum
+conversation](https://forum.polkadot.network/t/polkadot-assethub-high-nft-collection-deposit/4262),
+which provide critical perspectives on the implications of such changes.
+
+Adjusting NFT deposit requirements on Polkadot and Kusama Asset Hubs involves key challenges:
+
+1. **State Growth and Technical Concerns**: Lowering deposit requirements can lead to increased
+   blockchain state size, potentially causing state bloat. This growth needs to be managed to
+   prevent strain on the network's resources and maintain operational efficiency. As stated earlier,
+   the deposit levels proposed here are intentionally low with the thesis that future participants
+   would pay the standard rate.
+
+2. **Network Security and Market Response**: Adapting to the cryptocurrency market's volatility is
+   crucial. The mechanism for setting deposit amounts must be responsive yet stable, avoiding undue
+   complexity for users.
+
+3. **Economic Impact on Previous Stakeholders**: The change could have varied economic effects on
+   previous (before the change) creators, platform operators, and investors. Balancing these
+   interests is essential to ensure the adjustment benefits the ecosystem without negatively
+   impacting its value dynamics. However in the particular case of Polkadot and Kusama Asset Hub
+   this does not pose a concern since there are very few collections currently and thus previous
+   stakeholders wouldn't be much affected. As of date 9th January 2024 there are 42 collections on
+   Polkadot Asset Hub and 191 on Kusama Asset Hub with a relatively low volume.
+
+## Testing, Security, and Privacy
+
+### Security concerns
+
+As noted above, state bloat is a security concern. In the case of abuse, governance could adapt by
+increasing deposit rates and/or using `forceDestroy` on collections agreed to be spam.
+
+## Performance, Ergonomics, and Compatibility
+
+### Performance
+
+The primary performance consideration stems from the potential for state bloat due to increased
+activity from lower deposit requirements. It's vital to monitor and manage this to avoid any
```

Comment:
Do we have monitoring in place?

Reply:
Maybe @karim or someone from KodaDot is tracking NFT activity?

Reply:
I think @vikiival here is in a good position to answer this question, since he created indexer for AssetHub.

Reply:
> Someone from KodaDot is tracking NFT activity?

```sql
-- current nfts 9792
select count(*) from nft_entity where burned = false;

-- current collection 98
select count(*) from collection_entity where burned = false;

-- all nfts 9883 (~ 100 burned)
select count(*) from nft_entity;

-- all collections 138 (40 destroyed)
select count(*) from collection_entity;

-- all interactions that is not mint (list, buy, burn) 16_019
select count(*) from event where interaction <> 'MINT';
```

Top 3 creators on AHP
```json
  [{
    "count": 41,
    "issuer": "15CoYMEnJhhWHvdEPXDuTBnZKXwrJzMQdcMwcHGsVx5kXYvW"
  },
  {
    "count": 24,
    "issuer": "1ULZhwpUPLLg5VRYiq6rBHY8XaShAmBW7kqGBfvHBqrgBcN"
  },
  {
    "count": 6,
    "issuer": "1nYXo6fd1fjVm1ixUwycELfjTyaxxzVHAVgZ8S8H6cZMfY7"
  }]
  ```
  
Data are public via our @subsquid indexer or [Typescript API (kodadot/uniquery)](https://github.com/kodadot/uniquery)